sgkit/io/plink/plink_writer.py

```diff
@@ -78,11 +103,12 @@ def write_plink(
         "bp_position": ds.variant_position.values,
         "allele_1": ds.variant_allele.values[:, 0],
         "allele_2": ds.variant_allele.values[:, 1],
-        "iid": ds.sample_id,
+        "iid": ds.sample_id.values,  # may be overridden by sample_member_id below (if present)
     }
 
-    if "variant_id" in ds:
-        properties["sid"] = ds.variant_id
+    for var, prop in FAM_VARIABLE_TO_BED_READER.items():
+        if var in ds:
+            properties[prop] = ds[var].values
```

Comment:
I might be missing something but does the new/optional missing value `'.'` need to be transformed back to `0`?

Reply:
Done